meteor/client/ui/ActiveRundownView.tsx

```diff
@@ -1,159 +1,97 @@
-import * as React from 'react'
+import React, { useEffect } from 'react'
 import * as _ from 'underscore'
-import { Route, Switch } from 'react-router-dom'
-import { translateWithTracker, Translated } from '../lib/ReactMeteorData/ReactMeteorData'
-import { RundownPlaylist } from '../../lib/collections/RundownPlaylists'
+import { NavLink, Route, Switch, useRouteMatch } from 'react-router-dom'
+import { useSubscription, useTracker } from '../lib/ReactMeteorData/ReactMeteorData'
 
 import { Spinner } from '../lib/Spinner'
 import { RundownView } from './RundownView'
-import { MeteorReactComponent } from '../lib/MeteorReactComponent'
-import { objectPathGet } from '../../lib/lib'
 import { PubSub } from '../../lib/api/pubsub'
 import { UIStudios } from './Collections'
-import { UIStudio } from '../../lib/api/studios'
 import { StudioId } from '@sofie-automation/corelib/dist/dataModel/Ids'
 import { RundownPlaylists } from '../collections'
+import { useTranslation } from 'react-i18next'
 
-interface IProps {
-	match: {
-		params?: {
-			studioId: StudioId
-		}
-		path: string
-	}
-}
-interface ITrackedProps {
-	playlist?: RundownPlaylist
-	studio?: UIStudio
-	studioId?: StudioId
-	// isReady: boolean
-}
-interface IState {
-	subsReady: boolean
-}
-export const ActiveRundownView = translateWithTracker<IProps, {}, ITrackedProps>((props: IProps) => {
-	const studioId = objectPathGet(props, 'match.params.studioId')
-	let studio
-	if (studioId) {
-		studio = UIStudios.findOne(studioId)
-	}
-	const playlist = RundownPlaylists.findOne({
-		activationId: { $exists: true },
-		studioId: studioId,
-	})
+export function ActiveRundownView({ studioId }: { studioId: StudioId }): JSX.Element | null {
+	const { t } = useTranslation()
 
-	return {
-		playlist,
-		studio,
-		studioId,
-	}
-})(
-	class ActiveRundownView extends MeteorReactComponent<Translated<IProps & ITrackedProps>, IState> {
-		constructor(props) {
-			super(props)
-			this.state = {
-				subsReady: false,
-			}
-		}
-
-		componentDidMount(): void {
-			this.subscribe(
-				PubSub.rundownPlaylists,
-				_.extend(
-					{
-						activationId: { $exists: true },
-					},
-					this.props.studioId
-						? {
-								studioId: this.props.studioId,
-						  }
-						: {}
-				)
-			)
+	const { path } = useRouteMatch()
 
-			if (this.props.studioId) {
-				this.subscribe(PubSub.uiStudio, this.props.studioId)
-			}
+	const studioReady = useSubscription(PubSub.uiStudio, studioId)
+	const playlistReady = useSubscription(
+		PubSub.rundownPlaylists,
+		_.extend(
+			{
+				activationId: { $exists: true },
+			},
+			studioId
```

Comment:
the typings say that `studioId` will also be defined now, so this ternary and `_.extend` look unnecessary